### PR TITLE
added reuasable workflow for PRs with a code basis of golang

### PR DESCRIPTION
**Pull request by jvhenriksen**
the reuable workflow consists of
- building the docker image based on the docker-compose.production.yml
- check the linting of the golang code
- running the go tests

One example usage of this reusable workflow would be:

```
name: Check PR - Golang
on:
  pull_request:
    branches:
      - main

env:
  GOPRIVATE: github.com/hkraftno/byge-common

jobs:
  CI:
    name: Testing Go
    uses: hkraftno/.github/.github/workflows/byge-create-PR-go.yml@main
    with:
      go_module_path: ./service/.
      docker_path: ./service/.docker
    secrets: inherit

```

This yml file needs to be inside a private repository inside the `hkraftno` organization. It needs to be located in the following folder: `hkraftno/my-repo/.github/workflows`. When implemented, this workflow will run on pull requests to main, where it will run the above mentioned tests, found in the code in this branch. All service and job repos under `byge` should implement this workflow. 

By using this reuable workflow, we can easily change the workflow of all the repos, as this provides the source of truth of what the workflow should do

### CODE REVIEW

File: .github/workflows/byge-create-PR-go.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          # skip-cache: true
+
+          # Optional: if set to true, then the action won't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true, then the action won't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
+
+          # Optional: The mode to install golangci-lint. It can be 'binary' or 'goinstall'.
+          # install-mode: "goinstall"
```
Comment (FredrikMorstad) [inline .github/workflows/byge-create-PR-go.yml:87]: Are these necessary?

File: .github/workflows/byge-create-PR-go.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        run: |
+          cd ${{ inputs.docker_path }}
+          docker compose -f docker-compose.production.yml build
+
+      - name: Check if image built successfully
+        run: |
+          if [ $? -ne 0 ]; then
+            echo "Failed to build Docker image"
+            exit 1
+          fi
```
Comment (FredrikMorstad) [inline .github/workflows/byge-create-PR-go.yml:40]: Wouldn't the previous failed if the command exited with an error?

**Review by FredrikMorstad — CHANGES_REQUESTED** [review]
Left some comments that could be looked at

File: .github/workflows/byge-create-PR-go.yml (hunk 2 above)
Comment (jvhenriksen) [reply .github/workflows/byge-create-PR-go.yml:40]: seems like it is true what you said 👍

File: .github/workflows/byge-create-PR-go.yml (hunk 1 above)
Comment (jvhenriksen) [reply .github/workflows/byge-create-PR-go.yml:87]: 🔥

**Review by FredrikMorstad — APPROVED** [review]
LGTM, very good to have a shared workflow for all services 👍

**Pull request merged**